PowerShell Scanners/Battery Status/Battery Status.ps1

```diff
@@ -0,0 +1,22 @@
+﻿function Return-Battery($BatteryName = "No Battery Detected", $DesignedCapacity = 0, $FullCharge = 0, $Health = 0){
+    [PSCustomObject]@{
+        Name = $BatteryName
+        DesignedCapacity = $DesignedCapacity
+        FullCharge = $FullCharge
+        Health = $Health
+    }
+}
+
+
+$Batteries = (Get-WmiObject -Class "BatteryStatus" -Namespace "ROOT\WMI" -ErrorAction SilentlyContinue)
+$BatteryIndex = 0
+foreach($Battery in $Batteries){
+    $BatteryData = Get-WmiObject -Class "BatteryStaticData" -Namespace "ROOT\WMI"
```

Comment:
This seems out of place, is there a reason why it must be called for each battery retuned by `root\wmi\BatteryStatus` or could it be moved outside of the loop?

Reply:
Doh, you're right, that can definitely be moved out of the loop to avoid repeated WMI calls. I'll get those swapped over.